tests/raftstore/cluster.rs

```diff
@@ -146,19 +146,42 @@ impl<T: Simulator> Cluster<T> {
         self.sim.rl().call_command(request, timeout)
     }
 
+    fn naive_call_command_on_leader(&mut self,
+                                    region_id: u64,
+                                    mut request: RaftCmdRequest,
+                                    timeout: Duration)
+                                    -> Result<RaftCmdResponse> {
+        if let Some(leader) = self.leader_of_region(region_id) {
+            request.mut_header().set_peer(leader);
+            return self.call_command(request, timeout);
+        }
+        Err(box_err!("can't get leader of region"))
+    }
+
     pub fn call_command_on_leader(&mut self,
                                   region_id: u64,
-                                  mut request: RaftCmdRequest,
+                                  request: RaftCmdRequest,
                                   timeout: Duration)
                                   -> Result<RaftCmdResponse> {
-        for _ in 0..200 {
-            if let Some(leader) = self.leader_of_region(region_id) {
-                request.mut_header().set_peer(leader);
-                return self.call_command(request, timeout);
+        let mut retry_cnt = 0;
+        loop {
+            let result = self.naive_call_command_on_leader(region_id, request.clone(), timeout);
+            if let Ok(resp) = result {
+                if resp.get_header().has_error() && retry_cnt < 10 {
+                    retry_cnt += 1;
+                    if self.refresh_leader_if_needed(&resp, region_id) {
+                        warn!("seems leader changed, let's retry");
+                        continue;
+                    } else if resp.get_header().get_error().has_stale_epoch() {
```

Comment:
stale epoch checking is only ok for common request like get/put, not for admin command,